Theia application builds should be easy to reproduce

### Bug Description:

Theia is using "^1.x.y" instead of "1.x.y" in all modules, this causes two problems
1) Since romantic versioning is used within Theia repo, API breakage might be introduced between any version

2) It's harder to build applications using old versions of Theia extensions since if not every needed Theia extension is explicitly stated in dependencies/resolutions you will likely end up with pulling several parallel versions of @theia/core for example. Due to dependency injection we can not have multiple parallel versions of the same Theia extension module

### Steps to Reproduce:

1.  $ yo theia-extension:app none --browser 1 --electron 0 --extensionType empty --theia-version 1.3.0 --skip-install
2.  $ yarn
3.  Observe error:
ERROR in ../node_modules/@theia/monaco/lib/browser/monaco-frontend-module.js
Module not found: Error: Can't resolve '@theia/editor/lib/browser/semantic-highlight/semantic-highlighting-service' in '/local/home/<user>/dev/test/broken-again/node_modules
/@theia/monaco/lib/browser'

4. Observe that you have serveral different Theia extensions versions: $ yarn why @theia/core | grep Found
=> Found "@theia/core@1.9.0"
=> Found "none#@theia/core@1.3.0"
=> Found "browser-app#@theia/core@1.3.0"
=> Found "browser-app#@theia/editor#@theia/core@1.9.0"
=> Found "browser-app#@theia/filesystem#@theia/core@1.9.0"
=> Found "browser-app#@theia/markers#@theia/core@1.9.0"
=> Found "browser-app#@theia/navigator#@theia/core@1.9.0"
=> Found "browser-app#@theia/process#@theia/core@1.9.0"
=> Found "browser-app#@theia/workspace#@theia/core@1.9.0"

### Additional Information
See this discussion regarding the issue: https://github.com/eclipse-theia/theia/discussions/8445 - Versioning and Releases etc. 

The discussion suggests to use resolutions as a work-around, this shouldn't be needed if package.json uses specific versions "1.x.y" instead of "^1.x.y"

Work-around for @theia/core:
"resolutions": {​​
"@theia/core": "1.3.0"
}​​
However the work-around gives these warnings:
$ yarn why @theia/core | grep Found
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.4.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
warning Resolution field "@theia/core@1.3.0" is incompatible with requested version "@theia/core@^1.9.0"
=> Found "@theia/core@1.3.0"

- Theia Version: All that has been delivered so far.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ## v1.10.0
 
 - [plugin] added `createDeployQuickOpenItem` method to create `DeployQuickOpenItem` in order to make extension deploy command extensible [#8919] (https://github.com/eclipse-theia/theia/pull/8919)
+- [dependencies] updated to use fixed versions when publishing, `"x.y.z"` instead of `"^x.y.z"` in dependencies [#8880](https://github.com/eclipse-theia/theia/pull/8880)
 
 ## v1.9.0 - 16/12/2020
 

--- a/package.json
+++ b/package.json
@@ -68,7 +68,7 @@
     "rebuild:electron": "theia rebuild:electron",
     "rebuild:electron:debug": "DEBUG=electron-rebuild && yarn rebuild:electron",
     "publish": "yarn && yarn test && yarn publish:latest",
-    "publish:latest": "lerna publish && yarn publish:check",
+    "publish:latest": "lerna publish --exact && yarn publish:check",
     "publish:next": "yarn next:publish && yarn next:publish --skip-npm && yarn publish:check",
     "next:publish": "lerna publish --exact --canary=next --npm-tag=next --yes",
     "publish:check": "node scripts/check-publish.js",

